### ISSUE TITLE
GET requests for secrets are not issued in kubemark

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/kubernetes#36368]
While running tests on real clusters, I see a big number of requests about secretes. E.g. those are logs from 1 hour of the test run:
```
wojtekt@e2e-enormous-cluster-master ~ $ cat /var/log/kube-apiserver.log-20161107-1478527202 | grep GET | grep secrets | grep kubelet | wc -l
2899260
```
This gives almost 1000 requests per second.

Those requests all not present in kubemark at all.

We should investigate why it is the case and try to change kubemark to match production more (at least in this term).

**Comment by gmarek** [comment]
@dchen1107 @yujuhong - does and when Kubelet asks for secrets?

**Comment by wojtek-t** [comment]
it does, but from what I found only here:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L464

**Comment by gmarek** [comment]
This one looks like it should be done once for every Pod. Where those gets are coming from?

**Comment by yujuhong** [comment]
kubelet gets the secret for every pod on every sync.
There is an issue to open to cache the secrets: #19188

**Comment by wojtek-t** [comment]
@yujuhong - why this is not happening in kubemark? Do we do periodic sync for pods there?

**Comment by gmarek** [comment]
My guess is that our pods don't have any secrets, but @wojtek-t question is valid - does SimpleKubelet does this?

**Comment by yujuhong** [comment]
I haven't looked at the SimpleKubelet code for a long time, but as long as you are not using RunOnce kubelet, there should be periodic sync. 

> My guess is that our pods don't have any secrets,

This is possible. Most pods have a default service account including an access token for the apiserver as the secret. I am not sure if this is the cause in the kubemark cluster.

**Comment by wojtek-t** [comment]
>
My guess is that our pods don't have any secrets, 

Marek - this is unrelated. We are running _exactly the same test__ and:
- in Kubemark we don't have those calls
- in real cluster we have those calls

That means - there is significant different between real cluster and kubemark which we don't wanto to have.

**Comment by gmarek** [comment]
Yes, but as @yujuhong mentioned it's possible that we don't create a service account for the master, which would end up with Pods not having those Secrets. I'll look into in next week-ish.

**Comment by gmarek** [comment]
@yujuhong, @dchen1107 - I tried adding ServiceAccounts to the Kubemark master, so the Secrets would be generated. This didn't work straight away, because I wasn't registering secret volume plugin. After I added that it still isn't working.

The reason why is, IIUC:
```
E1115 11:20:15.391159      10 mount_linux.go:119] Mount failed: exit status 32
Mounting command: mount
Mounting arguments: tmpfs /tmp/hollow-kubelet.735025627/pods/2f114e46-ab25-11e6-b7b7-42010af00002/volumes/kubernetes.io~secret/default-token-xm0en tmpfs []
Output: mount: permission denied
```
Am I right thinking that it's because containers can't create and mount tmpfs volumes? Can I pass some flag/option somewhere to make it allowed? Or is there some other easy (even extremely hacky) workaround?

This is the WIP PR that can be used to repro problems I'm having: #36810
